packages/@react-spectrum/dnd/docs/dnd.mdx

```diff
@@ -189,85 +182,283 @@ function DraggableList(props) {
+
+Due to the possible complexity of a droppable collection's behavior, `useDnDHooks` accepts various [high level drop event handlers and options](#high-level-drop-handlers-and-options) that help you enable common drop behaviors like
+reordering or item insertion operations. If you find that you need greater control over the drop behavior, you can use `onDrop` and `getDropOperation` instead as demonstrated [below](#low-level-api).
+
+For a more general explanation of these low level options, see the `useDroppableCollection` aria [docs](/react-aria/useDroppableCollection.html#low-level-api).
+
+#### Using high level options
+
+For the second ListView we want to make it droppable, accepting drops between items and onto folders but not at the root of the list. Similar to the draggable ListView, we'll start by initializing the default item list
 via [useListData](/react-stately/useListData.html). As a reminder, `useListData` is completely optional here and can be replaced by [useAsyncList](/react-stately/useAsyncList.html) or any other state management solution.
 
 ```tsx
 let list = useListData({
   initialItems: [
-    {id: 'f', textValue: 'Adobe AfterEffects'},
-    {id: 'g', textValue: 'Adobe Illustrator'},
-    {id: 'h', textValue: 'Adobe Lightroom'},
-    {id: 'i', textValue: 'Adobe Premiere Pro'},
-    {id: 'j', textValue: 'Adobe Fresco'}
+    {id: 'f', type: 'file', name: 'Adobe AfterEffects'},
+    {id: 'g', type: 'file', name: 'Adobe Illustrator'},
+    {id: 'h', type: 'file', name: 'Adobe Lightroom'},
+    {id: 'i', type: 'file', name: 'Adobe Premiere Pro'},
+    {id: 'j', type: 'file', name: 'Adobe Fresco'},
+    {id: 'k', type: 'folder', name: 'Apps',  childNodes: []}
   ]
 });
 ```
 
-To implement root level only drops, we create a `getDropOperation` function that returns `"cancel"` for any drop targets other than `root` or if the dragged item types doesn't include
-`'adobe-app'`.
+Since we only want to accept items from the first ListView, we setup an array containing our custom type that will be provided to `useDnDHooks`'s `acceptedDragTypes` option.
+
+```tsx
+let acceptedDragTypes = ['adobe-app'];
+```
+
+To handle the processing of the drop item data, we create a `itemProcessor` utility function that will be used in each of the drop handlers. See the [aria docs](/react-aria/useDroppableCollection.html#drop-data)
+for more information on the drop items provided to the drop handlers.
 
 ```tsx
-function getDropOperation(target, types) {
-  if (target.type !== 'root' || !types.has('adobe-app')) {
-    return 'cancel';
+let itemProcessor = async (items, acceptedDragTypes) => {
+  let processedItems = [];
+  let text;
+  for (let item of items) {
+    for (let type of acceptedDragTypes) {
+      if (item.kind === 'text' && item.types.has(type)) {
+        text = await item.getText(type);
+        processedItems.push(JSON.parse(text));
+        break;
+      }
+    }
   }
```

Comment:
Yeah, it isn't technically necessary for this example, I'll go ahead and change it. I liked having the `itemProcessor` since it would be easier to expand on if the user wanted to handle directories/files as well or have separate logic for Android Chrome DnD like we have in our stories

Reply:
Another reason is to accommodate for typescript since the items in the drop handlers are of type `DropItem[]` which includes Files and Directories. Unfortunately the type won't reflect the filtering that has already occurred in useDroppableCollection